firmware/src/machine.c

```diff
@@ -0,0 +1,231 @@
+#include "machine.h"
+#include "usart.h"
+#include "pwm.h"
+
+
+volatile control_t control;
+volatile error_flags_t error_flags;
+volatile state_machine_t state_machine;
+void machine_init(void)
+{
+    TCCR2A  =   (1 << WGM21) | (0 << WGM20)         // Timer 2 in Mode 2 = CTC (clear on compar  e)
+            | (0 << COM2A1) | (0 << COM2A0)         // do nothing with OC2A
+            | (0 << COM2B1) | (0 << COM2B0);        // do nothing with OC2B
+    TCCR2B  =   (0 << WGM22)                        // Timer 0 in Mode 2 = CTC (clear on compar  e)
+            | (0 << FOC0A) | (0 << FOC0B)           // dont force outputs
+            | (1 << CS22)                           // clock enabled, prescaller = 1024
+            | (1 << CS21)
+            | (1 << CS20);
+    OCR2A   =   66;                              // Valor para igualdade de comparacao A par  a frequencia de 150 Hz
+    TIMSK2 |=   (1 << OCIE2A);                      // Ativa a interrupcao na igualdade de comp  aração do TC2 com OCR2A
+	//TODO revisar isso
+} 
+
+void check_buffers(void)
+{
+    //VERBOSE_MSG_MACHINE(usart_send_string("Checking buffers..."));
+    while(!CBUF_IsFull(cbuf_adc0));
+    while(!CBUF_IsFull(cbuf_adc1));
+    while(!CBUF_IsFull(cbuf_adc2));
```

Comment:
dangerous code! can enter in a infinite loop if you don't consume the buffer fast enough